### PR TITLE
Use TypeShape to proxy requests

### PR DESCRIPTION
**Pull request by kerams**
Makes `DynamicRecord` (apart from `makeDocsSchema`), `AsyncBoxer` and a good portion of each webserver adapter obsolete.

Todo:
- [ ] Take http verbs into account `| ("GET" | "POST"), SingleArgument(input, _) when input = typeof<unit>` (is it really necessary though?)
- [ ] Implement new adapters for ASP.NET and Giraffe
- [ ] Improve error messages for when the shape of the request or API record is not valid
- [ ] Clean up (Types.fs and TypeInfo.fs appear to be unused)

### CODE REVIEW

File: Fable.Remoting.Suave/FableSuaveAdapter.fs [hunk 1]
```diff
@@ -72,108 +69,51 @@ module SuaveUtil =
...
-            | SingleArgument (i, t) when t = typeof<Async<byte[]>> -> true
-            | ManyArguments (i, t) when t = typeof<Async<byte[]>> -> true
-            | otherwise -> false
-             
-          let isFableProxyRequest = 
-            context.request.headers
-            |> Map.ofList
-            |> Map.containsKey "x-remoting-proxy" 
-
-          if isBinaryOutput && isFableProxyRequest then
```
[the remark this answers is not in the document]
Comment (Zaid-Ajaj) [reply Fable.Remoting.Suave/FableSuaveAdapter.fs:95]: Backwards compatibility with Fable.Remoting.Client versions that didn't support this option. Though the general recommendation has always been to use latest server-side component with latest Fable.Remoting.Client

[the remark this answers is not in the document]
Comment (kerams) [reply Fable.Remoting.Suave/FableSuaveAdapter.fs:95]: I see. I've ported it too, so it's included if you want to keep it.

[the remark this answers is not in the document]
Comment (Zaid-Ajaj) [reply Fable.Remoting.Suave/FableSuaveAdapter.fs:95]: yeah let's keep it in for now :smile:

**Comment by kerams** [comment]
Turns out the .NET client does not send `x-remoting-proxy` and as a result `byte[]` responses are serialized with JSON instead of sending raw bytes as is the case with the Fable client.

I've also changed the MsgPack MIME type to `application/msgpack` based on https://github.com/msgpack/msgpack/issues/194, so that it's possible to control compression.

**Comment by Zaid-Ajaj** [comment]
Hi @kerams, I take that this is ready for merge?

**Comment by kerams** [comment]
If you have no objections, then yes ;)

**Comment by Zaid-Ajaj** [comment]
Published all the goodies 🚀 removed .ionide folder and added local dotnet tools to the project

**Comment by kerams** [comment]
It's still in the repo 😄

**Comment by Zaid-Ajaj** [comment]
LOL I just deleted it *and* removed it from .gitignore so that the deletion gets picked up :joy: OK let me try again

**Comment by Zaid-Ajaj** [comment]
OK it *should* be gone now :smile:

**Comment by kerams** [comment]
Weird build failures though.
